app/acapy.js

```diff
@@ -0,0 +1,86 @@
+const { ACAPY_API_KEY, ACAPY_ADMIN_SERVICE } = require('./env')
+
+const querySubWallets = async (subwalletName) => {
+  const fetch = (await import('node-fetch')).default
+
+  const url = new URL(`${ACAPY_ADMIN_SERVICE}/multitenancy/wallets`)
+  const search = new URLSearchParams({ wallet_name: subwalletName })
+  url.search = search.toString()
+
+  const queryResponse = await fetch(url, {
+    headers: {
+      'x-api-key': ACAPY_API_KEY,
+    },
+  })
+
+  const wallets = await queryResponse.json()
+
+  return wallets.results
+}
+
+const createSubWallet = async (body) => {
+  const fetch = (await import('node-fetch')).default
+
+  const url = new URL(`${ACAPY_ADMIN_SERVICE}/multitenancy/wallet`)
+  const createResponse = await fetch(url, {
+    method: 'POST',
+    headers: {
+      'x-api-key': ACAPY_API_KEY,
+      'content-type': 'application/json',
+    },
+    body: JSON.stringify(body),
+  })
+
+  return createResponse
+}
+
+const getSubWalletToken = async (subWalletName) => {
+  const fetch = (await import('node-fetch')).default
+
+  const wallets = await querySubWallets(subWalletName)
+  if (wallets.length !== 1) {
+    return null
+  }
+
+  const wallet = wallets[0]
+  const url = new URL(`${ACAPY_ADMIN_SERVICE}/multitenancy/wallet/${wallet.wallet_id}/token`)
+  const subWalletToken = await fetch(url, {
+    method: 'POST',
+    headers: {
+      'x-api-key': ACAPY_API_KEY,
+      'content-type': 'application/json',
+    },
+    body: JSON.stringify({}),
+  })
+
+  const token = (await subWalletToken.json()).token
+  return token
+}
+
+const noBodyMethods = new Set(['GET', 'HEAD'])
+const subWalletCall = async (token, { path, query, method, body }) => {
+  const fetch = (await import('node-fetch')).default
+
+  const url = new URL(`${ACAPY_ADMIN_SERVICE}${path}`)
+  const search = new URLSearchParams(query)
```

Comment:
not sure if we want an extra module, but this is a great one qs -> https://www.npmjs.com/package/qs

Reply:
interesting. Wpould there be any particular advantage (apart one one line of code saved) of using this over native `URLSearchParams`?